ui/StatusQ/src/submodelproxymodel.cpp

```diff
@@ -50,15 +52,40 @@ QVariant SubmodelProxyModel::data(const QModelIndex &index, int role) const
         context->setContextProperty(QStringLiteral("submodel"), submodel);
 
         QObject* instance = m_delegateModel->create(context);
+
         instance->setParent(submodelObj);
 
         QVariant wrappedInstance = QVariant::fromValue(instance);
 
+        QObject* connector = m_connector->createWithInitialProperties(
+                    { { "target", QVariant::fromValue(instance) } });
+
+        instance->setParent(submodelObj);
+        connector->setParent(connector);
+
+        connect(connector, SIGNAL(customRoleChanged(QObject*,int)),
+                this, SLOT(onCustomRoleChanged(QObject*,int)));
+
         submodelObj->setProperty(attachementPropertyName, wrappedInstance);
 
         return QVariant::fromValue(wrappedInstance);
     }
 
+    if (role >= m_additionalRolesOffset
+            && role < m_additionalRolesOffset + m_additionalRolesMap.size())
+    {
+        auto submodel = data(index, m_submodelRole);
+
+        auto submodelObj = submodel.value<QObject*>();
+
+        if (submodelObj == nullptr) {
+            qWarning("Submodel must be a QObject-based type!");
+            return QVariant{};
+        }
+
+        return submodelObj->property(m_roleNames[role] + "Role");
```

Comment:
```suggestion
        return submodelObj->property(QByteArrayLiteral("%1Role").arg(m_roleNames[role]);
```

Reply:
there is no `arg()` on `QByteArray`. What can be done is this:
```suggestion
        return submodelObj->property(m_roleNames[role]
                                     + QByteArrayLiteral("Role"));
```

Reply:
Ouch, what did I think there is? :thinking: